ground/gcs/src/plugins/pathplanner/flightdatamodel.cpp

```diff
@@ -534,28 +531,45 @@ void FlightDataModel::readFromFile(QString fileName)
             data = new PathPlanData;
             double wpLLA[3];
             double wpNED[3];
+            int params = 0;
             while (!fieldNode.isNull()) {
                 QDomElement field = fieldNode.toElement();
                 if (field.tagName() == "field") {
-                    if(field.attribute("name") == "down")
+                    if(field.attribute("name") == "down") {
                         wpNED[2] = field.attribute("value").toDouble();
-                    else if(field.attribute("name") == "description")
+                        params++;
+                    } else if(field.attribute("name") == "description") {
                         data->wpDescription=field.attribute("value");
-                    else if(field.attribute("name") == "north")
+                        params++;
+                    } else if(field.attribute("name") == "north") {
                         wpNED[0] = field.attribute("value").toDouble();
-                    else if(field.attribute("name") == "east")
+                        params++;
+                    } else if(field.attribute("name") == "east") {
                         wpNED[1] = field.attribute("value").toDouble();
-                    else if(field.attribute("name") == "velocity")
+                        params++;
+                    } else if(field.attribute("name") == "velocity") {
                         data->velocity = field.attribute("value").toFloat();
-                    else if(field.attribute("name") == "mode")
+                        params++;
+                    } else if(field.attribute("name") == "mode") {
                         data->mode=field.attribute("value").toInt();
-                    else if(field.attribute("name") == "mode_params")
+                        params++;
+                    } else if(field.attribute("name") == "mode_params") {
                         data->mode_params = field.attribute("value").toFloat();
-                    else if(field.attribute("name") == "is_locked")
+                        params++;
+                    } else if(field.attribute("name") == "is_locked") {
                         data->locked = field.attribute("value").toInt();
+                        params++;
+                    }
                 }
                 fieldNode=fieldNode.nextSibling();
             }
+
+            // We can't really check everything in the file is fully consistent, but
+            // at least we can make sure we have the right number of fields
+            if (params < 8) {
```

Comment:
Will this catch if a field is duplicated?  Might make sense to have ceilings on north/east/down/velocity, too.